### DISCUSSION TITLE
How can I ingest WAF logs when the bucket uses Kinesis Firehose namespaces?

### DISCUSSION THREAD

**Discussion by alexzon-tr** [aws-solutions/centralized-logging-with-opensearch#121] [Q&A] [answered]
I use the following AWS solution for WAF implementation:
https://aws.amazon.com/solutions/implementations/security-automations-for-aws-waf/

The solution sends WAF logs to a Firehose and the Firehose sends them to an S3 Bucket using the following prefix:
`AWSLogs/year=!{timestamp:yyyy}/month=!{timestamp:MM}/day=!{timestamp:dd}/hour=!{timestamp:HH}/`

When creating the Log ingestion on the Centralized Logging solution, it detects this prefix and sets the `logBucketPrefix` parameter on the Cloudformation Stack.

The log ingestion **doesn't** work and I think the prefix may be the reason.

I tried to search for a clue in the Lambda logs without success. I couldn't understand how all the Functions work together.

**Comment by James96315** [comment, marked as answer]
You can choose the `Manual Method` to setup.
![image](https://github.com/aws-solutions/centralized-logging-with-opensearch/assets/75407126/78e4afa4-0563-441a-9599-2182ff5765d3)

**Reply by alexzon-tr** [reply]
Worked like a charm!
Thank you so much.
